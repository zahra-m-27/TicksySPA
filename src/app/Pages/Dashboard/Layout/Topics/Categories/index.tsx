import {message, Spin} from 'antd';
import Category from './Category';
import API from '../../../../../API';
import styles from './styles.module.scss';
import Assets from '../../../../../Assets';
import React, {useEffect, useState} from 'react';
import {useHistory, useParams} from 'react-router-dom';
import TButton from '../../../../../Components/TButton';
import TopicDto from '../../../../../API/DTOs/TopicDto';
import CategoryDto from '../../../../../API/DTOs/CategoryDto';
import TPagination from '../../../../../Components/TPagination';

export default function Categories() {
  const history = useHistory();
  const params = useParams<{topicId: string}>();

  const [Total, setTotal] = useState(1);
  const [Topic, setTopic] = useState<TopicDto>();
  const [PageSize, setPageSize] = useState(100);
  const [PageNumber, setPageNumber] = useState(0);
  const [Categories, setCategories] = useState<CategoryDto[]>();

  const getData = () => {
    API.Topics.GetTopicCategories({
      limit: PageSize,
      offset: PageNumber * PageSize,
      topicId: parseInt(params.topicId),
    })
      .then((response) => {
        setTotal(response.count);
        setCategories(response.results);
      })
      .catch(() => {
        message.error('امکان گرفتن دسته بندی های این تاپیک وجود ندارد');
      });
    API.Topics.GetTopic({
      topicId: parseInt(params.topicId),
    }).then((response) => setTopic(response));
  };

  const onDelete = (categoryId: number) => {
    API.Topics.DeleteCategory({
      categoryId: categoryId,
      topicId: parseInt(params.topicId),
    })
      .then(() => {
        getData();
        message.success('دسته بندی حذف شد');
      })
      .catch(() => {
        message.error('خطایی در حذف دسته بندی رخ داده است');
      });
  };

  useEffect(getData, []);

  if (!Topic || !Categories) {
    return <Spin size="large" />;
  }

  let content = (
    <div className={styles.not_data}>
      <Assets.SVGs.SadSvg />
      <span className={styles.not_data_text}>دسته بندی ای یافت نشد</span>

      <TButton
        backgroundColor="#1e77c9"
        label="دسته بندی جدید ایجاد کن"
        className={styles.create_no_data}
        labelClassName={styles.create_no_data_label}
        onClick={() => history.push(`/dashboard/topics/${params.topicId}/new`)}
      />
    </div>
  );

  if (Categories.length)
    content = (
      <>
        <div className={styles.categories}>
          {Categories.map((category, key) => (
            <Category
              key={key}
              category={category}
              topicId={parseInt(params.topicId)}
              onClick={() => history.push('/dashboard/tickets/' + category.id)}
              onDelete={() => onDelete(category.id)}
            />
          ))}
        </div>
        <TPagination
          className={styles.pagination}
          total={Total}
          pageSize={PageSize}
          pageNumber={PageNumber + 1}
          onChange={(value) => setPageNumber(value - 1)}
        />
      </>
    );

  return (
    <div className={styles.container}>
      <div className={styles.topic}>
        <img src={Topic.avatar} alt="topic-avatar" />
        <span>{Topic.title}</span>

        <TButton
          label="ایجاد دسته بندی"
          className={styles.create_category}
          labelClassName={styles.create_category_label}
          onClick={() =>
            history.push(`/dashboard/topics/${params.topicId}/new`)
          }
        />
      </div>
      {content}
    </div>
  );
}
