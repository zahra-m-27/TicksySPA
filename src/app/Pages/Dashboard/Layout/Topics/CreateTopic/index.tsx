import API from '../../../../../API';
import {Button, message} from 'antd';
import Assets from '../../../../../Assets';
import {useRef, useState} from 'react';
import styles from './styles.module.scss';
import {useHistory} from 'react-router-dom';
import SEInput from '../../../../../Components/SEInput';

export default function CreateTopic() {
  const history = useHistory();
  const [Title, setTitle] = useState('');
  const [Avatar, setAvatar] = useState<File>();
  const [Loading, setLoading] = useState(false);
  const [AvatarUrl, setAvatarUrl] = useState<any>();
  const usernameRef = useRef<HTMLInputElement>(null);
  const [Description, setDescription] = useState('');
  const descriptionRef = useRef<HTMLTextAreaElement>(null);
  const [TitleHasError, setTitleHasError] = useState(false);
  const [DescriptionHasError, setDescriptionHasError] = useState(false);

  const onTitleEnter = () => {
    if (usernameRef && usernameRef.current) {
      usernameRef.current.focus();
    }
  };
  const onUsernameEnter = () => {
    if (descriptionRef && descriptionRef.current) {
      descriptionRef.current.focus();
    }
  };

  const onCreateTopic = () => {
    if (!Title) {
      setTitleHasError(true);
    }
    if (!Description) {
      setDescriptionHasError(true);
    }
    if (!Avatar) {
      message.error('برای تاپیک خود یک آواتار انتخاب کنید');
    }
    if (!Title || !Description || !Avatar) {
      return;
    }
    setTitleHasError(false);
    setDescriptionHasError(false);

    setLoading(true);
    API.Topics.CreateTopic({
      title: Title,
      avatar: Avatar,
      description: Description,
    })
      .then(() => {
        message.success('تاپیک با موفقیت ایجاد شد.');
        history.push('/dashboard/topics');
      })
      .catch((error) => {
        if (error.status === 403) {
          message.error('برای این عملیات نیاز هست تا احراز هویت کرده باشید');
        } else {
          message.error('انجام این عملیات ممکن نیست');
        }
      })
      .finally(() => setLoading(false));
  };

  return (
    <div className={styles.container}>
      <div className={styles.rectangle}>
        <div className={styles.header}>
          تاپيک جدید
          <img src={Assets.SVGs.Topic} alt="" />
        </div>
        <div className={styles.upload}>
          <label htmlFor="picture">
            <img
              alt=""
              className={styles.picture}
              src={AvatarUrl ?? Assets.Images.GoogleImage2}
            />
          </label>
          <input
            type="file"
            id="picture"
            className={styles.upload_image}
            onChange={(e) => {
              if (e.target.files) {
                setAvatar(e.target.files[0]);
                const fr = new FileReader();
                fr.onload = function () {
                  setAvatarUrl(fr.result);
                };
                fr.readAsDataURL(e.target.files[0]);
              }
            }}
          />
        </div>
        <div className={styles.middle}>
          <label>:عنوان</label>
          <SEInput
            content={Title}
            onEnter={onTitleEnter}
            onChangeText={setTitle}
            hasError={TitleHasError}
            innerContainerClassName={styles.input}
          />
        </div>
        <div className={styles.below}>
          <label>:توضيح</label>
          <SEInput
            minLines={5}
            ref={descriptionRef}
            content={Description}
            onChangeText={setDescription}
            hasError={DescriptionHasError}
            innerContainerClassName={styles.input}
          />
        </div>
        <Button
          loading={Loading}
          type="primary"
          className={styles.enter_button}
          onClick={onCreateTopic}>
          ثبت
        </Button>
      </div>
    </div>
  );
}
