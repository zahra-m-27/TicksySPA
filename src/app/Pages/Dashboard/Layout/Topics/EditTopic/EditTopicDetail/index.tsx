import {message, Spin} from 'antd';
import API from '../../../../../../API';
import Assets from '../../../../../../Assets';
import styles from './styles.module.scss';
import {useEffect, useRef, useState} from 'react';
import SEInput from '../../../../../../Components/SEInput';
import {useHistory, useParams} from 'react-router-dom';
import EditTopicCard from '../../../../../../Components/EditTopicCard';

export default function EditTopicDetail() {
  const history = useHistory();
  const params = useParams<{topicId: string}>();
  const [Avatar, setAvatar] = useState<File>();
  const [Title, setTitle] = useState('');
  const [AvatarUrl, setAvatarUrl] = useState<any>();
  const [Description, setDescription] = useState('');
  const [SaveLoading, setSaveLoading] = useState(false);
  const descriptionRef = useRef<HTMLTextAreaElement>(null);
  const [TitleHasError, setTitleHasError] = useState(false);
  const [Loading, setLoading] = useState(!process.env.JEST_WORKER_ID);
  const [DescriptionHasError, setDescriptionHasError] = useState(false);

  useEffect(() => {
    setLoading(!process.env.JEST_WORKER_ID);
    API.Topics.GetTopic({
      topicId: parseInt(params.topicId),
    })
      .then((response) => {
        setTitle(response.title);
        setAvatarUrl(response.avatar);
        setDescription(response.description);
      })
      .catch(() => message.error('اشکالی در دریافت اطلاعات تایپک رخ داده.'))
      .finally(() => setLoading(false));
  }, []);

  const onEditTopic = () => {
    if (!Title) {
      setTitleHasError(true);
    }
    if (!Description) {
      setDescriptionHasError(true);
    }
    if (!Avatar && !AvatarUrl) {
      message.error('برای تاپیک خود یک آواتار انتخاب کنید');
    }
    if (!Title || !Description || !Avatar) {
      return;
    }
    setTitleHasError(false);
    setDescriptionHasError(false);

    setSaveLoading(true);
    API.Topics.UpdateTopic({
      title: Title,
      avatar: Avatar,
      description: Description,
      topicId: parseInt(params.topicId),
    })
      .then(() => {
        message.success('تاپیک با بروز شد.');
        history.push('/dashboard/topics');
      })
      .catch((error) => {
        if (error.status === 403) {
          message.error('برای این عملیات نیاز هست تا احراز هویت کرده باشید');
        } else {
          message.error('انجام این عملیات ممکن نیست');
        }
      })
      .finally(() => setSaveLoading(false));
  };

  const onTitleEnter = () => {
    if (descriptionRef && descriptionRef.current) {
      descriptionRef.current.focus();
    }
  };

  if (Loading) {
    return (
      <div className={styles.container}>
        <Spin size="large" />
      </div>
    );
  }

  return (
    <div className={styles.container}>
      <EditTopicCard
        title="ویرایش تاپيک"
        icon={Assets.SVGs.Topic}
        className={styles.detail_card}
        buttons={[
          {
            label: 'ثبت',
            testId: 'submit',
            onClick: onEditTopic,
            loading: SaveLoading,
            className: styles.enter_button,
          },
        ]}>
        <div className={styles.upload}>
          <label htmlFor="picture">
            <img
              alt="avatar"
              className={styles.picture}
              src={AvatarUrl ?? Assets.Images.GoogleImage2}
            />
          </label>
          <input
            type="file"
            id="picture"
            data-testid="avatar"
            className={styles.upload_image}
            onChange={(e) => {
              if (e.target.files) {
                setAvatar(e.target.files[0]);
                const fr = new FileReader();
                fr.onload = function () {
                  setAvatarUrl(fr.result);
                };
                fr.readAsDataURL(e.target.files[0]);
              }
            }}
          />
        </div>
        <div className={styles.inputs_container}>
          <div className={styles.middle}>
            <label>عنوان</label>
            <SEInput
              content={Title}
              data-testid="title"
              onEnter={onTitleEnter}
              onChangeText={setTitle}
              hasError={TitleHasError}
              innerContainerClassName={styles.input_container}
            />
          </div>
          <div className={styles.below}>
            <label>توضيح</label>
            <SEInput
              minLines={5}
              ref={descriptionRef}
              content={Description}
              data-testid="description"
              onChangeText={setDescription}
              hasError={DescriptionHasError}
              inputClassName={styles.description_input}
              innerContainerClassName={styles.input_container}
            />
          </div>
        </div>
      </EditTopicCard>
    </div>
  );
}
