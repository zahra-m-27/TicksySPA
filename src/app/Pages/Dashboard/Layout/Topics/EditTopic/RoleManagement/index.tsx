import {message, Spin} from 'antd';
import API from '../../../../../../API';
import styles from './styles.module.scss';
import {useParams} from 'react-router-dom';
import Assets from '../../../../../../Assets';
import React, {MutableRefObject, useEffect, useRef, useState} from 'react';
import TButton from '../../../../../../Components/TButton';
import showDialog from '../../../../../../Components/TDialog';
import ManageTopicRoleDialog from '../../../../../../Dialogs/ManageTopicRole';
import TPagination from '../../../../../../Components/TPagination';
import TopicAdminListItemDto from '../../../../../../API/DTOs/TopicAdminListItemDto';

interface Props {
  onUpdate: MutableRefObject<(() => void) | undefined>;
}

export default function RoleManagement({onUpdate}: Props) {
  const dismissDialog = useRef<() => void>();
  const params = useParams<{topicId: string}>();

  const [Total, setTotal] = useState(1);
  const [PageSize, setPageSize] = useState(5);
  const [PageNumber, setPageNumber] = useState(0);
  const [Roles, setRoles] = useState<TopicAdminListItemDto[]>();

  const getData = () => {
    API.Topics.GetTopicRoles({
      limit: PageSize,
      offset: PageNumber * PageSize,
      topicId: parseInt(params.topicId),
    })
      .then((response) => {
        setTotal(response.count);
        setRoles(
          response.results.concat(
            new Array(
              (PageSize - (response.results.length % PageSize)) % PageSize
            ).fill({})
          )
        );
      })
      .catch(() => {
        message.error('خطایی در گرفتن نقش های تاپیک رخ داده.');
      });
  };

  onUpdate.current = getData;
  useEffect(getData, [PageNumber]);

  let content = <Spin size="large" />;

  if (Roles)
    content = (
      <div className={styles.not_data}>
        <Assets.SVGs.SadSvg />
        <span className={styles.not_data_text}>دسته بندی ای یافت نشد</span>

        <TButton
          backgroundColor="#1e77c9"
          label="نقش جدید ایجاد کن"
          className={styles.create_no_data}
          labelClassName={styles.create_no_data_label}
          onClick={() =>
            (dismissDialog.current = showDialog({
              content: (
                <ManageTopicRoleDialog
                  onUpdate={getData}
                  onDismissRef={dismissDialog}
                  topicId={parseInt(params.topicId)}
                />
              ),
            }))
          }
        />
      </div>
    );

  if (Roles?.length)
    content = (
      <>
        <table className={styles.table}>
          <thead>
            <tr>
              <th>ردیف</th>
              <th>عنوان نقش</th>
              <th>تعداد مسئولان</th>
              <th>عملیات</th>
            </tr>
          </thead>
          <tbody>
            {Roles.map((role, index) => {
              if (!role.id) return <tr />;
              return (
                <tr key={index}>
                  <td>{role.id}</td>
                  <td>{role.title}</td>
                  <td>{role.users_detail.length}</td>
                  <td>
                    <img
                      alt="edit"
                      src={Assets.Images.EditUsers}
                      className={styles.edit_icon}
                      onClick={() =>
                        (dismissDialog.current = showDialog({
                          content: (
                            <ManageTopicRoleDialog
                              roleId={role.id}
                              onUpdate={getData}
                              title={role.title}
                              onDismissRef={dismissDialog}
                              topicId={parseInt(params.topicId)}
                              users={role.users_detail.map((u) => u.id)}
                            />
                          ),
                        }))
                      }
                    />
                    <img
                      alt="delete"
                      onClick={() => {
                        API.Topics.DeleteRole({
                          roleId: role.id,
                          topicId: parseInt(params.topicId),
                        })
                          .then(() => {
                            message.success(`نقش ${role.title} حذف شد.`);
                          })
                          .catch(() => {
                            message.error('خطایی در حذف نقش ها رخ داده.');
                          })
                          .finally(getData);
                      }}
                      src={Assets.Images.Delete}
                      className={styles.delete_icon}
                    />
                  </td>
                </tr>
              );
            })}
          </tbody>
        </table>
        <TPagination
          total={Total}
          pageSize={PageSize}
          pageNumber={PageNumber + 1}
          onChange={(value) => setPageNumber(value - 1)}
        />
      </>
    );

  return <div className={styles.container}>{content}</div>;
}
