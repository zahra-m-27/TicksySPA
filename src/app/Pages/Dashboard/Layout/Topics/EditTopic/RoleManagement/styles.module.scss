$border_color: #cccccc;
.container {
  flex: 1;
  display: flex;
  align-items: center;
  flex-direction: column;
}

.header {
  width: 98%;
  display: flex;
  padding: 15px 10px;
  flex-direction: row;
  border-radius: 15px;
  align-items: center;
  justify-content: space-between;
  border: 1px solid #cccccc;
  box-shadow: 0 1px 1px 0 #cccccc, 0 0 4px 0 #cccccc2e;
}

.mode_box {
  background-color: #e5e9ef;
  width: 32.7%;
  height: 75%;
  border-radius: 8px;
  margin-right: 1.1%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
}

.select_mode {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 44%;
  height: 83.3%;
  border-radius: 5px;
  transition: all 0.2s;
  cursor: pointer;
  color: #606f7b;
  font-size: clamp(12px, 1vw, 20px);
}

.not_data {
  flex: 1;
  display: flex;
  margin-bottom: 20%;
  align-items: center;
  flex-direction: column;
  justify-content: center;

  svg {
    width: 85px;
    height: 85px;
    fill: #5ea3d0;
  }
  .not_data_text {
    font-size: 20px;
    margin-top: 30px;
    font-weight: 600;
  }
}

.create {
  &_no_data {
    margin-top: 20px;

    &_label {
      color: white;
      font-size: 18px;
      font-weight: 600;
    }
  }
}

.table {
  width: 100%;
  table-layout: auto;
  text-align: center;
  border-spacing: 0 1.8vh;
  border-collapse: separate;

  th {
    border: none;
    color: white;
    background-color: #2168bd;

    &:first-of-type {
      border-radius: 0 10px 10px 0;
    }

    &:last-of-type {
      border-radius: 10px 0 0 10px;
    }
  }

  tr {
    height: 70px;
  }

  td {
    border-left: 0;
    border-right: 0;
    font-size: 16px;
    border-top: 1px solid $border_color;
    border-bottom: 1px solid $border_color;

    &:first-child {
      border-radius: 0 5px 5px 0;
      border-right: 1px solid $border_color;
    }

    &:last-child {
      border-radius: 5px 0 0 5px;
      border-left: 1px solid $border_color;
    }
  }
}

.edit_icon {
  cursor: pointer;
  margin-left: 0.7vw;
}
.delete_icon {
  cursor: pointer;
}
