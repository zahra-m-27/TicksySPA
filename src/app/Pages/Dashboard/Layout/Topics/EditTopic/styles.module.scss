.container {
  flex: 1;
  padding: 25px 50px;
}

.add_button {
  padding: 5px 10px;
  margin-bottom: 5px;

  span {
    font-size: clamp(12px, 1vw, 16px);
  }
}
