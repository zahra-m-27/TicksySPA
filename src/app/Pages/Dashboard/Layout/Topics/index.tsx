import API from '../../../../API';
import TopicCard from '../../../../Components/TopicCard';
import Assets from '../../../../Assets';
import styles from './styles.module.scss';
import {useEffect, useState} from 'react';
import {useHistory} from 'react-router-dom';
import TopicListItemDto from '../../../../API/DTOs/TopicListItemDto';
import {message} from 'antd';

export default function Topics() {
  const history = useHistory();
  const [Topics, setTopics] = useState<TopicListItemDto[]>([]);

  useEffect(() => {
    API.Topics.GetTopics({
      limit: 100,
      offset: 0,
    })
      .then((response) => {
        setTopics(
          response.results
            .filter((a) => !Topics.find((b) => a.id === b.id))
            .concat(Topics)
        );
      })
      .catch((error) => {
        if (error.status === 403) {
          message.error('برای این عملیات نیاز هست تا احراز هویت کرده باشید');
        } else {
          message.error('انجام این عملیات ممکن نیست');
        }
      });
  }, []);

  return (
    <div className={styles.container}>
      <div
        className={styles.add_topic}
        onClick={() => history.push('/dashboard/topics/new')}>
        <p>افزودن</p>
        <img src={Assets.Images.Add} alt="" />
      </div>
      <div className={styles.cards}>
        {Topics.map((item, index) => (
          <TopicCard
            key={index}
            topicId={item.id}
            title={item.title}
            avatar={item.avatar}
            description={item.description}
          />
        ))}
      </div>
    </div>
  );
}
