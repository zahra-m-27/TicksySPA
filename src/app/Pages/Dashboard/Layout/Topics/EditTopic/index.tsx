import styles from './styles.module.scss';
import {Button, Tabs} from 'antd';
import React, {useRef, useState} from 'react';
import EditTopicDetail from './EditTopicDetail';
import UserManagement from './UserManagement';
import RoleManagement from './RoleManagement';
import showDialog from '../../../../../Components/TDialog';
import ManageTopicRoleDialog from '../../../../../Dialogs/ManageTopicRole';
import TButton from '../../../../../Components/TButton';
import ManageTopicUserDialog from '../../../../../Dialogs/ManageTopicUser';
import {useParams} from 'react-router-dom';

export default function EditTopic() {
  const updateRoles = useRef<() => void>();
  const updateUsers = useRef<() => void>();
  const dismissDialog = useRef<() => void>();
  const params = useParams<{topicId: string}>();
  const [Tab, setTab] = useState('detail');

  let addTitle = '';
  if (Tab === 'roles') {
    addTitle = 'افزودن نقش';
  } else if (Tab === 'users') {
    addTitle = 'افزودن کاربر';
  }

  return (
    <div className={styles.container}>
      <Tabs
        accessKey={Tab}
        onChange={setTab}
        tabBarExtraContent={
          addTitle && {
            right: (
              <TButton
                label={addTitle}
                className={styles.add_button}
                onClick={() => {
                  if (Tab === 'roles')
                    dismissDialog.current = showDialog({
                      content: (
                        <ManageTopicRoleDialog
                          onDismissRef={dismissDialog}
                          onUpdate={updateRoles.current}
                          topicId={parseInt(params.topicId)}
                        />
                      ),
                    });
                  else
                    dismissDialog.current = showDialog({
                      content: (
                        <ManageTopicUserDialog
                          onDismissRef={dismissDialog}
                          onUpdate={updateUsers.current}
                          topicId={parseInt(params.topicId)}
                        />
                      ),
                    });
                }}
              />
            ),
          }
        }>
        <Tabs.TabPane tab="مشخصات" key="detail">
          <EditTopicDetail />
        </Tabs.TabPane>
        <Tabs.TabPane tab="کاربران" key="users">
          <UserManagement onUpdate={updateUsers} />
        </Tabs.TabPane>
        <Tabs.TabPane tab="نقش ها" key="roles">
          <RoleManagement onUpdate={updateRoles} />
        </Tabs.TabPane>
      </Tabs>
    </div>
  );
}
