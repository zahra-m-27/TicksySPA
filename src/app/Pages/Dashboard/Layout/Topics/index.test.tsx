import ReactDOM from 'react-dom';
import React from 'react';
import {render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import EditTopic from './EditTopic/RoleManagement';
import RoleManagement from './EditTopic/RoleManagement';
import SubmitCertificate from '../Tickets';
import Ticket from '../Tickets';
import Tickets from '../Tickets';
import Topics from './index';

const sample = (
  <MemoryRouter>
    <Topics />
  </MemoryRouter>
);

test('render in ReactDOM', () => {
  const div = document.createElement('div');
  ReactDOM.render(sample, div);
});

it('snapshot', () => {
  const snapshot = renderer.create(sample).toJSON();
  expect(snapshot).toMatchSnapshot();
});
