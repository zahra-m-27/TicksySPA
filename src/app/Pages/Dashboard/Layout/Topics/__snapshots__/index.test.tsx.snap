// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="container"
>
  <div
    className="add_topic"
    onClick={[Function]}
  >
    <p>
      افزودن
    </p>
    <img
      alt=""
      src="add.png"
    />
  </div>
  <div
    className="cards"
  />
</div>
`;
